extension/visual.lua

```diff
@@ -1,89 +1,196 @@
+		
 		if info then
-			height = info.height or height
-			r = info.r or r
-			g = info.g or g
-			b = info.b or b
-			a = info.a or a
-			writeZ = info.writeZ ~= nil and info.writeZ or writeZ
-			additive = info.additive ~= nil and info.additive or additive
+			r = info.r and info.r or 1
+			g = info.g and info.g or 1
+			b = info.b and info.b or 1
+			a = info.a and info.a or 1
+			width = info.width or width
 			target = info.target or target
+			if info.writeZ ~= nil then writeZ = info.writeZ end
+			if info.additive ~= nil then additive = info.additive end
+			DrawFunction = info.DrawFunction ~= nil and info.DrawFunction or (info.writeZ == false and DebugLine or DrawLine)
+		end
+
+		if sprite then
+			local middle = VecScale(VecAdd(source, destination), .5)
+			local len = VecLength(VecSub(source, destination))
+			local transform = Transform(middle, QuatRotateQuat(QuatLookAt(source, destination), QuatEuler(-90, 0, 0)))
+			local target_local = TransformToLocalPoint(transform, target)
+			target_local[2] = 0
+			local transform_fixed = TransformToParentTransform(transform, Transform(nil, QuatLookAt(target_local, nil)))
+
+			DrawSprite(sprite, transform_fixed, width, len, r, g, b, a, writeZ, additive)
+		else
+			DrawFunction(source, destination, r, g, b, a);
 		end
+    end
+    function visual.drawlines(sprites, sources, connect, info)
+        sprites = type(sprites) ~= "table" and {sprites} or sprites
 
-		DrawSprite(sprite, Transform(pos, QuatLookAt(pos, target)), size, height, r, g, b, a, writeZ, additive)
+        for i = 1, #sprites do
+            local sourceCount = #sources
+
+            for j = 1, sourceCount - 1 do
+                visual.drawline(sprites[i], sources[j], sources[j + 1], info)
+            end
+
+            if connect then
+                visual.drawline(sprites[i], sources[1], sources[sourceCount], info)
+            end
+        end
 	end
 
-	function visual.drawaxis(transform, rot, mul, writeZ)
-		mul = mul or 1
-		if not transform.pos then transform = Transform(transform, rot or QUAT_ZERO) end
-		local DrawFunc = writeZ and DrawLine or DebugLine
-		DrawFunc(transform.pos, TransformToParentPoint(transform, Vec(mul, 0, 0)), 1, 0, 0)
-		DrawFunc(transform.pos, TransformToParentPoint(transform, Vec(0, mul, 0)), 0, 1, 0)
-		DrawFunc(transform.pos, TransformToParentPoint(transform, Vec(0, 0, mul)), 0, 0, 1)
+	function visual.drawaxis(transform, quat, radius, writeZ)
+		scale = scale or 1
+		if not transform.pos then transform = Transform(transform, quat or QUAT_ZERO) end
+		local DrawFunction = writeZ and DrawLine or DebugLine
+
+		DrawFunction(transform.pos, TransformToParentPoint(transform, Vec(radius, 0, 0)), 1, 0, 0)
+		DrawFunction(transform.pos, TransformToParentPoint(transform, Vec(0, radius, 0)), 0, 1, 0)
+		DrawFunction(transform.pos, TransformToParentPoint(transform, Vec(0, 0, radius)), 0, 0, 1)
+    end
+    
+    function visual.drawpolygon(transform, quat, radius, rotation, sides, info)
```

Comment:
unlike `visual.drawaxis`, having `quat` here is a problem since it is followed by non-optional parameters. It would probably be best to just keep `transform` and discard the option to pass a vector and a quaternion separately.